agenta-backend/agenta_backend/services/llm_apps_service.py

```diff
@@ -14,25 +14,59 @@
 logger.setLevel(logging.DEBUG)
 
 
-def extract_result_from_response(response):
+def extract_result_from_response(response: dict):
+    def get_nested_value(d: dict, keys: list, default=None):
+        """
+        Helper function to safely retrieve nested values.
+        """
+
+        for key in keys:
+            if isinstance(d, dict):
+                d = d.get(key, default)
+            else:
+                return default
+        return d
+
+    # Initialize default values
     value = None
     latency = None
     cost = None
 
-    if response.get("version", None) == "2.0":
+    # Handle version 3.0 response
+    if response.get("version") == "3.0":
         value = response
+        # Ensure 'data' is a dictionary or convert it to a string
+        if not isinstance(value.get("data"), dict):
+            value["data"] = str(value.get("data"))
 
-        if not isinstance(value["data"], dict):
-            value["data"] = str(value["data"])
+        if "trace" in response:
+            trace_tree = (
```

Comment:
have you tried a RAG evaluator to see if the trace works the same ?

Reply:
Yup, RAG evaluators works for v2.0 and v3.0. 😉